simtools/util/general.py

```diff
@@ -801,43 +802,54 @@ def convert_2D_to_radial_distr(xaxis, yaxis, hist2d, bin_size=50, max_dist=1000)
     np.array
         The values of the 1D histogram with size = int(max_dist/bin_size).
     """
+    logger = logging.getLogger(__name__)
+    # Check if the histogram will make sense
+    bins_step = (
+        2 * max_dist / num_bins
+    )  # in the 2D array, the positive and negative direction count.
+    warn = False
```

Comment:
Instead of this `warn` you can simply register the warning inside the for loop and break. You save this non-elegant variable, an `if` call and going through the second axis if the first one already is problematic.

Reply:
But you will still fill the histogram. The `break` will just stop the for loop of the warning, not stop running all together.

Reply:
I see your point now. indeed, it is more elegant, thanks.